android/src/main/java/com/swmansion/rnscreens/utils/ScreenDummyLayoutHelper.kt

```diff
@@ -179,11 +179,16 @@ internal class ScreenDummyLayoutHelper(
         }
 
         val topLevelDecorView = requireActivity().window.decorView
+        val topInset = DecorViewInsetsUtils.getDecorViewTopInset(topLevelDecorView)
 
-        // These dimensions are not accurate, as they do include status bar & navigation bar, however
+        // These dimensions are not accurate, as they do include navigation bar, however
         // it is ok for our purposes.
         val decorViewWidth = topLevelDecorView.width
-        val decorViewHeight = topLevelDecorView.height
+        // Adjust the height of the top-level decor view by subtracting the top inset.
+        // This accounts for edge-to-edge support where we need to manually do frame correction
+        // for AppBarLayout. To ensure proper layout for the Screen, we manually downscale the content
+        // to accommodate the extra padding.
+        val decorViewHeight = topLevelDecorView.height - topInset
```

Comment:
Does this impact the layout result in any way? It seems to me, that the computed header height would be independent of the `decorViewHeight` as long as `decorViewHeight` is high enough and it does not limit the header.

Reply:
https://github.com/software-mansion/react-native-screens/pull/3442/commits/13bded578c6e0d8734120ea982135a4efc3d79f1